src/pcre2_substitute.c

```diff
@@ -832,8 +1235,15 @@ do
             /* Peek backwards one code unit to obtain the length of the mark.
             It can (theoretically) contain an embedded NUL. */
             fraglength = mark[-1];
-            CHECKMEMCPY(mark, fraglength);
+            if (forcecase.to_case != PCRE2_SUBSTITUTE_CASE_NONE &&
+                substitute_case_callout == NULL)
+              CHECKCASECPY(mark, fraglength);
+            else
+              CHECKMEMCPY(mark, fraglength);
             }
+
+            // XXX ^^^ This is a behavioural change (to do casing on a MARK)
+            // Ensure we have test coverage, and check it's OK with Philip
```

Comment:
@PhilipHazel I made a behavioural change here. Previously if you did `\U${*MARK}` it would copy the mark without case-transforming it.

I decided this was probably an oversight, rather than a conscious decision, so I fixed it.

Reply:
I assume that it does already case transform /U${1} etc,? I think it certainly ought to, that's the whole point. So yes, I agree.

Reply:
Yes, copying fragments from the replacement string, or from the subject string, were being case-forced. It was just the special case of copying the mark which was using the original casing of the mark.

I just wanted to check, rather than silently "fix" something.